src/main/java/com/mongodb/jdbc/BsonTypeInfo.java

```diff
@@ -1,342 +1,230 @@
-                return 1;
-            case BINDATA_TYPE_NAME:
-            case STRING_TYPE_NAME:
-            case ARRAY_TYPE_NAME:
-            case OBJECT_TYPE_NAME:
-            case OBJECTID_TYPE_NAME:
-            case DBPOINTER_TYPE_NAME:
-            case JAVASCRIPT_TYPE_NAME:
-            case JAVASCRIPTWITHSCOPE_TYPE_NAME:
-            case MAXKEY_TYPE_NAME:
-            case MINKEY_TYPE_NAME:
-            case REGEX_TYPE_NAME:
-            case SYMBOL_TYPE_NAME:
-            case TIMESTAMP_TYPE_NAME:
-            case UNDEFINED_TYPE_NAME:
-            case NULL_TYPE_NAME:
-            case BSON_TYPE_NAME:
-                return null;
+    /**
+     * Converts a bson type name to a BsonTypeInfo
+     *
+     * @param typeName is the bson type string as returned from the mongodb aggregation $type
+     *     function, this list of viable inputs is in the BSON_TYPE_NAMES static Set above.
+     * @return BsonTypeInfo object corresponding to bson type name.
+     */
+    public static BsonTypeInfo getBsonTypeInfoByName(String typeName) throws SQLException {
+        if (!BSON_TYPE_NAMES.contains(typeName)) {
+            throw new SQLException("Unknown bson type name: \"" + typeName + "\"");
         }
-        throw new SQLException("unknown bson typeName: " + typeName);
-    }
 
-    public static int getJDBCType(String typeName) throws SQLException {
-        switch (typeName) {
-            case ARRAY_TYPE_NAME:
-            case DBPOINTER_TYPE_NAME:
-            case OBJECT_TYPE_NAME:
-            case JAVASCRIPT_TYPE_NAME:
-            case JAVASCRIPTWITHSCOPE_TYPE_NAME:
-            case MAXKEY_TYPE_NAME:
-            case MINKEY_TYPE_NAME:
-            case OBJECTID_TYPE_NAME:
-            case REGEX_TYPE_NAME:
-            case SYMBOL_TYPE_NAME:
-            case TIMESTAMP_TYPE_NAME:
-            case UNDEFINED_TYPE_NAME:
-                return Types.OTHER;
-            case BINDATA_TYPE_NAME:
-                return Types.BINARY;
-            case BOOL_TYPE_NAME:
-                return Types.BIT;
-            case DATE_TYPE_NAME:
-                return Types.TIMESTAMP;
-            case INT_TYPE_NAME:
-                return Types.INTEGER;
-            case LONG_TYPE_NAME:
-                return Types.BIGINT;
-            case NULL_TYPE_NAME:
-                return Types.NULL;
-            case STRING_TYPE_NAME:
-                return Types.LONGVARCHAR;
-            case DECIMAL_TYPE_NAME:
-                return Types.DECIMAL;
-            case DOUBLE_TYPE_NAME:
-                return Types.DOUBLE;
+        // All type names can be determined uniquely off a combination of first letter
+        // and length except for "minKey" vs "maxKey" and "string" vs "symbol".
+        switch (typeName.charAt(0)) {
+            case 'a':
+                return BSON_ARRAY;
+            case 'b':
+                switch (typeName.length()) {
+                    case 4:
+                        return BSON_BOOL;
```

Comment:
Adding the BSON type to the switch. 

```suggestion
                    case 4:
                        switch (typeName.charAt(1)) {
                            case 'o':
                                return BSON_BOOL;
                            case 's':
                                return BSON_BSON;
```

Reply:
no changes required; see comment above